> Hello @trichner thanks for the new device support.
> 
> Can you please also add it to the smoketests here? https://github.com/tinygo-org/drivers/blob/dev/smoketest.sh

Done!
Though running them fails on some other test locally:
```
> make test
...
src/machine/board_bluepill.go:6:2: package device/stm32 is not in GOROOT (<redacted>/.cache/tinygo/goroot-3db76b2bbe6512878c9fc4ff18b9674eba0696a404ae4e2915298e4f7a087fad/src/device/stm32)
```

--- a/examples/sht4x/main.go
+++ b/examples/sht4x/main.go
@@ -10,7 +10,7 @@ import (
 
 func main() {
 	machine.I2C0.Configure(machine.I2CConfig{})
-	sensor := sht4x.New(machine.I2C0, sht4x.DefaultAddress)
+	sensor := sht4x.New(machine.I2C0)
 
 	for {
 		temp, humidity, _ := sensor.ReadTemperatureHumidity()

--- a/sht4x/sht4x.go
+++ b/sht4x/sht4x.go
@@ -18,18 +18,15 @@ const (
 // Device represents a SHT4x sensor
 type Device struct {
 	bus     drivers.I2C
-	address uint8
+	Address uint8
 }
 
 // New creates a new SHT4x connection. The I2C bus must already be
 // configured.
-func New(bus drivers.I2C, addr uint8) Device {
-	if addr == 0 {
-		addr = DefaultAddress
-	}
+func New(bus drivers.I2C) Device {
 	return Device{
 		bus:     bus,
-		address: addr,
+		Address: DefaultAddress,
 	}
 }
 
@@ -53,7 +50,7 @@ func (d *Device) ReadTemperatureHumidity() (temperatureMilliCelsius int32, relat
 
 // rawReadings returns the sensor's raw values of the temperature and humidity
 func (d *Device) rawReadings() (uint16, uint16, error) {
-	err := d.bus.Tx(uint16(d.address), []byte{commandMeasurement}, nil)
+	err := d.bus.Tx(uint16(d.Address), []byte{commandMeasurement}, nil)
 	if err != nil {
 		return 0, 0, err
 	}
@@ -62,7 +59,7 @@ func (d *Device) rawReadings() (uint16, uint16, error) {
 	time.Sleep(10 * time.Millisecond)
 
 	var data [6]byte
-	err = d.bus.Tx(uint16(d.address), nil, data[:])
+	err = d.bus.Tx(uint16(d.Address), nil, data[:])
 	if err != nil {
 		return 0, 0, err
 	}

